test/nn/test_linear_cross_entropy.py

```diff
@@ -0,0 +1,256 @@
+        )
+        ref = _reference_linear_cross_entropy(
+            input_clone,
+            weight_clone,
+            target,
+            bias_clone,
+            reduction=reduction,
+            ignore_index=ignore_index,
+            label_smoothing=label_smoothing,
+        )
+
+        if fused.requires_grad:
+            grad_args = [
+                tensor for tensor in (input, weight, bias) if tensor is not None
+            ]
+            grad_args_ref = [
+                tensor
+                for tensor in (input_clone, weight_clone, bias_clone)
+                if tensor is not None
+            ]
+
+            if reduction == "none":
+                grad_output = torch.ones_like(fused)
+                fused_grads = torch.autograd.grad(
+                    fused,
+                    grad_args,
+                    grad_outputs=grad_output,
+                    retain_graph=False,
+                    allow_unused=True,
+                )
+                ref_grads = torch.autograd.grad(
+                    ref,
+                    grad_args_ref,
+                    grad_outputs=grad_output,
+                    retain_graph=False,
+                    allow_unused=True,
+                )
+            else:
+                fused_grads = torch.autograd.grad(
+                    fused,
+                    grad_args,
+                    retain_graph=False,
+                    allow_unused=True,
+                )
+                ref_grads = torch.autograd.grad(
+                    ref,
+                    grad_args_ref,
+                    retain_graph=False,
+                    allow_unused=True,
+                )
+
+            for grad_fused, grad_ref, tensor in zip(fused_grads, ref_grads, grad_args):
+                if grad_fused is None or grad_ref is None:
+                    self.assertTrue(grad_fused is None and grad_ref is None)
+                else:
+                    self.assertEqual(grad_fused, grad_ref)
+
+        if reduction == "none":
+            self.assertEqual(fused.shape, target.shape)
+            self.assertEqual(ref.shape, target.shape)
+        self.assertEqual(fused, ref)
+
+    def test_forward_backward_matches_reference_auto(self) -> None:
+        torch.manual_seed(0)
+        input = torch.randn(2, 3, 32, requires_grad=True)
+        weight = torch.randn(6000, 32, requires_grad=True)
+        bias = torch.randn(6000, requires_grad=True)
+        target = torch.randint(0, 6000, (2, 3))
+        self._compare_with_reference(
+            input, weight, target, bias, chunking_strategy="auto"
+        )
+
+    def test_vocab_chunking(self) -> None:
+        torch.manual_seed(0)
+        input = torch.randn(4, 16, requires_grad=True)
+        weight = torch.randn(5000, 16, requires_grad=True)
+        target = torch.randint(0, 5000, (4,))
+        self._compare_with_reference(
+            input, weight, target, None, chunking_strategy="vocab"
+        )
```

Comment:
@krocki With the new OpInfo implementation, you should be able to remove all the tests in this file that do `_compare_with_reference`.